File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/DSP-DSL/spectralAnalysis.py
def main() {

	var input = getRangeOfVector(0, 50000, 0.000125);
        var fft_real = fft1dreal(input);
        var fft_img = fft1dimg(input);
        var sq_abs = square(fft_real) + square(fft_img)  ;
        var sum1 = sum(sq_abs);
        var len1 = len(input);
        var res = sum1 / len1;
        print(res);
}

File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/Matlab/spectralAnalysis.m
% Constants
INPUT_LENGTH = 100000000;

% getRange function
input = getRange(0, INPUT_LENGTH, 1);

% DFT function (using built-in FFT)
fft_result = fft(input);

% Square of absolute values
sq_abs = abs(fft_result).^2;

% Sum and average
res = mean(sq_abs);

% Display result
fprintf('%f\n', res);

%  getRange function
function output = getRange(start, noOfSamples, increment)
    output = start + (0:noOfSamples-1) * increment;
end